From b03a34935d75b1d4277095c0a25dc9a8cbfe01ff Mon Sep 17 00:00:00 2001
From: yxj-github-437 <2457369732@qq.com>
Date: Tue, 4 Aug 2026 01:16:31 +0800
Subject: [PATCH] disable d when build gettext

---
 Makefile.def |  2 +-
 Makefile.in  | 20 ++++++++++----------
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/Makefile.def b/Makefile.def
index 549763de837..ce0124882f2 100644
--- a/Makefile.def
+++ b/Makefile.def
@@ -80,7 +80,7 @@ host_modules= { module= gettext; bootstrap=true; no_install=true;
 		// need it in some configurations, which is determined via nontrivial tests.
 		// Always enabling pic seems to make sense for something tied to
 		// user-facing output.
-		extra_configure_flags='--disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf';
+		extra_configure_flags='--disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf';
 		missing= pdf;
 		missing= html;
 		missing= info;
diff --git a/Makefile.in b/Makefile.in
index f34142cc077..6dd303e94af 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -19990,7 +19990,7 @@ configure-gettext:
 	  $$s/$$module_srcdir/configure \
 	  --srcdir=$${topdir}/$$module_srcdir \
 	  $(HOST_CONFIGARGS) --build=${build_alias} --host=${host_alias} \
-	  --target=${target_alias} --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf \
+	  --target=${target_alias} --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf \
 	  || exit 1
 @endif gettext
 
@@ -20026,7 +20026,7 @@ configure-stage1-gettext:
 	  --target=${target_alias} \
 	   \
 	  $(STAGE1_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stage2-gettext maybe-configure-stage2-gettext
@@ -20060,7 +20060,7 @@ configure-stage2-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGE2_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stage3-gettext maybe-configure-stage3-gettext
@@ -20094,7 +20094,7 @@ configure-stage3-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGE3_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stage4-gettext maybe-configure-stage4-gettext
@@ -20128,7 +20128,7 @@ configure-stage4-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGE4_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stageprofile-gettext maybe-configure-stageprofile-gettext
@@ -20162,7 +20162,7 @@ configure-stageprofile-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEprofile_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stagetrain-gettext maybe-configure-stagetrain-gettext
@@ -20196,7 +20196,7 @@ configure-stagetrain-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEtrain_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stagefeedback-gettext maybe-configure-stagefeedback-gettext
@@ -20230,7 +20230,7 @@ configure-stagefeedback-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEfeedback_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stageautoprofile-gettext maybe-configure-stageautoprofile-gettext
@@ -20264,7 +20264,7 @@ configure-stageautoprofile-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEautoprofile_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 .PHONY: configure-stageautofeedback-gettext maybe-configure-stageautofeedback-gettext
@@ -20298,7 +20298,7 @@ configure-stageautofeedback-gettext:
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEautofeedback_CONFIGURE_FLAGS) \
-	  --disable-shared --disable-threads --disable-java --disable-csharp --with-pic --disable-libasprintf
+	  --disable-shared --disable-threads --disable-d --disable-java --disable-csharp --with-pic --disable-libasprintf
 @endif gettext-bootstrap
 
 
-- 
2.53.0

